core/src/main/java/jenkins/I18n.java

```diff
@@ -93,7 +93,18 @@ public HttpResponse doResourceBundle(StaplerRequest request) {
         String language = request.getParameter("language");
         String country = request.getParameter("country");
         String variant = request.getParameter("variant");
-
+        // https://www.w3.org/International/questions/qa-lang-priorities
+        // in case we have regions/countries in the language query parameter
+        if (country == null && language != null) {
+            String[] languageTokens = language.split("-|_");
+            language = languageTokens[0];
+            if (languageTokens.length > 1) {
+                country = languageTokens[1];
+                if (languageTokens.length > 2) {
+                    variant = languageTokens[2];
```

Comment:
🐛 it overrides variant even if it is explicitly specified by the request field. The field should have higher priority to be consistent with <code>content</code>